src/history.ts

```diff
@@ -43,13 +43,14 @@ export class History {
   // Event handlers
 
   onPopState = (event: PopStateEvent) => {
-    if (this.shouldHandlePopState()) {
-      const { turbolinks } = event.state
-      if (turbolinks) {
-        const location = Location.currentLocation
-        const { restorationIdentifier } = turbolinks
-        this.delegate.historyPoppedToLocationWithRestorationIdentifier(location, restorationIdentifier)
-      }
+    if (!this.shouldHandlePopState()) return
+    if (!event.state) return
+
+    const { turbolinks } = event.state
+    if (turbolinks) {
+      const location = Location.currentLocation
+      const { restorationIdentifier } = turbolinks
+      this.delegate.historyPoppedToLocationWithRestorationIdentifier(location, restorationIdentifier)
     }
```

Comment:
Let's keep the existing no-early-return style or use it consistently:
```suggestion
    const { turbolinks } = event.state
    if (!turbolinks) return
    
    const location = Location.currentLocation
    const { restorationIdentifier } = turbolinks
    this.delegate.historyPoppedToLocationWithRestorationIdentifier(location, restorationIdentifier)
```

Reply:
👍